ci/docker/docker-compose.yml

```diff
@@ -0,0 +1,242 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# We use the cache_from feature introduced in file form version 3.4 (released 2017-11-01)
```

Comment:
Seems we don't have `docker-compose.yml` in v1.x currently. Do we need to backport this file?

Reply:
The docker-compose is used for setting up the templated Dockerfiles (same Dockerfile, but support different base images with or without nvidia drivers). It has better caching support than the previous `COPY` then `RUN` implementation and may be easier to maintain. It's possible to backport it. You can decide

Reply:
@leezu since we are not back porting every change back to v1.x from master. I think we can skip it. wdyt ?
If we want to templatize it in this PR we may have to remove most of dockerfiles here: https://github.com/apache/incubator-mxnet/tree/v1.x/ci/docker